### PR TITLE
Ignore isJoda flag from 7x nodes

### PR DESCRIPTION
**Pull request by pgomulka**
when upgrading from 7.7+ ES will send out a flag indicating if a pattern is of joda style. This is only used to support joda style indices in 7.x, in 8 we no longer support this.  All indices in 8 should use java style pattern. Hence we can ignore this flag. Similarly when writing from v8 to v7.7+ we should always send false flag.

relates https://github.com/elastic/elasticsearch/pull/52555

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/DocValueFormat.java [hunk 1]
```diff
@@ -213,6 +221,13 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeString(formatter.pattern());
             out.writeString(timeZone.getId());
             out.writeVInt(resolution.ordinal());
+            if (out.getVersion().onOrAfter(Version.V_7_7_0) && out.getVersion().before(Version.V_8_0_0)) {
+                /* when serializing to 7.7+  send out a flag indicating if a pattern is of joda style
+                   This is only used to support joda style indices in 7.x, in 8 we no longer support this.
+                   All indices in 8 should use java style pattern. Hence this flag is always true.
+                */
+                out.writeBoolean(true);
```
Comment (jaymode) [inline server/src/main/java/org/elasticsearch/search/DocValueFormat.java]: I think this should be false

Comment (pgomulka) [reply server/src/main/java/org/elasticsearch/search/DocValueFormat.java]: yes :)

**Review by jaymode — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/search/DocValueFormat.java [hunk 2]
```diff
@@ -201,6 +202,13 @@ public DateTime(StreamInput in) throws IOException {
             String zoneId = in.readString();
             this.timeZone = ZoneId.of(zoneId);
             this.resolution = DateFieldMapper.Resolution.ofOrdinal(in.readVInt());
+            if (in.getVersion().onOrAfter(Version.V_7_7_0) && in.getVersion().before(Version.V_8_0_0)) {
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/search/DocValueFormat.java]: I think the `in.getVersion().onOrAfter(Version.V_7_7_0)` can be removed, but I can understand keeping it for clarity and safety.

File: server/src/main/java/org/elasticsearch/search/DocValueFormat.java [hunk 3]
```diff
@@ -213,6 +221,13 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeString(formatter.pattern());
             out.writeString(timeZone.getId());
             out.writeVInt(resolution.ordinal());
+            if (out.getVersion().onOrAfter(Version.V_7_7_0) && out.getVersion().before(Version.V_8_0_0)) {
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/search/DocValueFormat.java]: similar comment as before `out.getVersion().onOrAfter(Version.V_7_7_0)` I don't think is strictly necessary. But nothing wrong with keeping it.

**Review by benwtrent — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/search/DocValueFormat.java (hunk 2 above)
Comment (pgomulka) [reply server/src/main/java/org/elasticsearch/search/DocValueFormat.java]: you are right, we don't expect mix cluster to work with versions before 7.7. The intention feels cleaner with it to me. I will keep it

**Pull request merged**